sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -223,7 +204,7 @@ case class OptimizeSkewedJoin(conf: SQLConf) extends Rule[SparkPlan] {
             shuffleStage.id == right.id =>
             PartialShuffleReaderExec(shuffleStage, skewedPartitions.toSet)
         }
-        subJoins += optimizedSmj
+        subJoins += optimizedSmj.asInstanceOf[SortMergeJoinExec].copy(partialSMJ = Some(true))
```

Comment:
the SMJ may not be the root node. Now I think it's better to match SMJ in the plan transformation above.